src/platform/packages/shared/kbn-unified-tabs/src/components/tabbed_content/tabbed_content.tsx

```diff
@@ -143,7 +144,28 @@ export const TabbedContent: React.FC<TabbedContentProps> = ({
       maxItemsCount,
       onDuplicate: (item) => {
         const newItem = createItem();
-        newItem.label = `${item.label} (copy)`;
+        const baseLabel = item.label.replace(/\s*\(copy\)( \d+)?$/, '');
+
+        // Find all existing copies to determine next number
+        const copyRegex = new RegExp(`^${baseLabel}\\s*\\(copy\\)( \\d+)?$`);
+        const copies = state.items
+          .filter((tab) => copyRegex.test(tab.label))
+          .map((tab) => {
+            const match = tab.label.match(/\(copy\) (\d+)$/);
+            return match && match[1] ? Number(match[1]) : 1; // match[1] is the number after (copy)
+          });
+
+        // Determine the next copy number
+        const nextNumber = copies.length > 0 ? Math.max(...copies) + 1 : null;
+
+        newItem.label = i18n.translate('unifiedTabs.duplicatedTabLabel', {
+          defaultMessage: nextNumber ? `{baseLabel} (copy) {number}` : '{baseLabel} (copy)',
+          values: {
+            baseLabel,
+            number: nextNumber,
+          },
```

Comment:
Our tools don't support conditional `defaultMessage`. We would need to define 2 separate `i18n.translate` or in this case we could define `number` variable as `nextNumber ? ` ${nextNumber}` : ''`.

Reply:
Thank you, wasn't aware of that!